### DISCUSSION TITLE
Is there a way to link to Play Store or App Store listing to prompt user to update?

### DISCUSSION THREAD

**Discussion by bchehraz** [ionic-team/capacitor#5326] [Q&A] [answered]
In general I'm trying to wire in some functionality where my server api can let the client know it has become incompatible. I want to tell the client if it's too old to go see if there's a new update available. Sometimes these updates may not be compatible, for instance, if they're on an old device. I can naively prompt them something along the lines of "hey, you're on an old client, go see if there are new updates [link to play store or app store]".

I've seen some apps have the ability to actually detect when an update is available on the Play Store and from within the app, trigger this update to happen without actually going to the Play Store. In an ideal world, this would be a great feature to have whenever I actually need the user to update their client-- but if this isn't possible then it's not the end of the world.

TLDR:

On an Ionic/Angular/Capacitor App built for iOS/Android...

1) Is it possible to link the app directly to the app store or play store listing of the app directly in the play store/app store app?

2) Is it possible to know whether there is an update available _prior_ to displaying this link in the app?

3) Is it possible to actually do the update within the app?

**Comment by robingenz** [comment, marked as answer]
Hi, I'll try to answer your questions.

1. Yes, this is possible on Android and iOS.
2. Yes, this is also possible on Android and iOS.
3. Yes, but this is only possible on Android (see https://developer.android.com/guide/playcore/in-app-updates).

You have to write a plugin for this or use an existing plugin (e.g. https://github.com/robingenz/capacitor-app-update).

I hope I could help you!

**Reply by bchehraz** [reply]
Awesome, thanks for the response. I'm installing your plugin now to see how it works

**Reply by bchehraz** [reply]
It's a little bit unclear on the section called "Android Variables" whether it requires any action from me... Do I have to add a new variable?

> ### Android Variables
> This plugin will use the following project variables (defined in your app’s variables.gradle file):
> - $androidPlayCore version of com.google.android.play:core (default: 1.9.0)

**Reply by robingenz** [reply]
No, you don't need to do anything at this point. This variable only gives you the option to change the used Android Play Core version.
Feel free to give me more feedback, I'm always happy about that.

If you decide to create your own plugin, you can take a look at the [Android](https://github.com/robingenz/capacitor-app-update/blob/main/android/src/main/java/dev/robingenz/capacitor/appupdate/AppUpdatePlugin.java) and [iOS](https://github.com/robingenz/capacitor-app-update/blob/main/ios/Plugin/AppUpdatePlugin.swift) code of the plugin. There you can see how the individual methods are implemented.

**Reply by bchehraz** [reply]
Thanks for the reply, I was able to get it working and it works pretty well so far. I'll have to weigh the pros & cons of creating my own, versus using your plugin as I don't know what the implications are.

While on the topic and seeing that you have some knowledge about interacting with the Play Store or App Store, do you know if there's any possible way to tell Play Store or App Store to _prevent_ an update from initiating if some condition is met? Say, the REST API I interact with is version 1.0, but as long as its 1.0, don't let any updates happen until it's a 2.0 (even if auto-updates are enabled in Play Store & App Store?).

**Reply by robingenz** [reply]
Glad to see that it works. 
I can only mention at this point that I am actively maintaining the plugin and will continue to do so for as long as possible.

> While on the topic and seeing that you have some knowledge about interacting with the Play Store or App Store, do you know if there's any possible way to tell Play Store or App Store to prevent an update from initiating if some condition is met? Say, the REST API I interact with is version 1.0, but as long as its 1.0, don't let any updates happen until it's a 2.0 (even if auto-updates are enabled in Play Store & App Store?).

As far as I know, there is unfortunately no option. When you release a new app version in the Dashboard, it is published and anyone can access it.
We usually do it this way: Release new API version, then release new app version. Both components are backward compatible for a limited time.

I hope I could help you.

**Reply by bchehraz** [reply]
Thank you for the explanation, that's also how I understand it... that there's no way to control it, but my boss insists there must be a way (lol) and I have yet to find any proof this is possible.

The particular issue we are trying to solve is that our customers can set up their own internal API servers, meaning that the same mobile client will be used by different customers with potentially different API versions depending on which one they set up. And those customer API servers are not forced to update. Meaning this can cause a mobile client to be _too new_ for any given server API.

**Reply by robingenz** [reply]
I know that. 😄  
One company I work with also has several customers with their own API servers but they also have their own (whitelabel) app. We then release app and server update at the same time.
So that would be one option. However, this would require whitelabel apps. 
This is fairly easy to achieve with Capacitor. Especially with [capacitor-configure](https://github.com/ionic-team/capacitor-configure) I see a lot of possibilities.
